device/thunder_ripple_sdk/src/processors/thunder_package_manager.rs

```diff
@@ -472,20 +491,28 @@ impl ThunderPackageManagerRequestProcessor {
         state: ThunderPackageManagerState,
         handle: String,
         operation: Operation,
+        status: Option<String>,
     ) {
-        if state
-            .active_operations
-            .lock()
-            .unwrap()
-            .remove(&handle)
-            .is_none()
-        {
-            state
-                .active_operations
-                .lock()
-                .unwrap()
-                .insert(handle.clone(), operation);
-            Self::start_operation_timer(state, handle);
+        /*
+        clone active_operations, we want an immutable view
+        */
+        let active_operations = state.active_operations.lock().unwrap().clone();
+        match active_operations.get(&handle) {
+            Some(op) => {
+                let mut timer = op.timer.clone();
+                timer.stop();
+                timer.insert_tag("status".to_string(), status.unwrap_or("".to_string()));
+                rdk_telemetry_emit(timer);
+            }
+            None => {
+                /*new operation, insert into map */
+                state
```

Comment:
Instead of locking again here can you just use `active_operations` and don't clone it on line 499?